lib/Core/Field/GenericResolvedFieldType.php

```diff
@@ -145,6 +145,10 @@ public function getOptionsResolver(): OptionsResolver
      */
     protected function newField($name, array $options): FieldConfig
     {
+        if ('@' === $name[0]) {
```

Comment:
@sstok does this seem correct to you? I'm kind of lost in the complexity, TBH. :see_no_evil:

Reply:
What do you mean exactly? The CS?

Reply:
No, the way I've done it here. It looks a bit smelly, sniffing the name everywhere.